mlserver/repository/repository.py

```diff
@@ -65,3 +76,17 @@ async def find(self, name: str) -> List[ModelSettings]:
             raise ModelNotFound(name)
 
         return selected
+
+
+class ModelRepositoryFactory:
+    @staticmethod
+    def resolve_model_repository(settings: Settings) -> ModelRepository:
+        result: ModelRepository
+        if settings.model_repository_implementation:
+            result = settings.model_repository_implementation(
+                **settings.model_repository_implementation_args
+            )
```

Comment:
To respect the current interface (and the current `model_repository_root` setting), would it make sense to always pass the root folder as a first argument, then followed by `model_repository_implementation_args`?

Reply:
Well, I was thinking about how to avoid a compatibility issue with the signature of the default and other modelRepository classes. I mean, now SchemalessModelRepository expects model_repository_root from settings, and on the other hand, model_repository_implementation expects parameters from settings. So how do conciliate both interfaces? I think I shouldn't make a breaking change with the previous implementation. I can write the code but please could you give the expected behavior with the default implementation (previous implementation) and the new one? 
My two cents:
```
if settings.model_repository_root:
  model_repository_implementation_args["file"] = settings.model_repository_root

```
That way, we could handle the default model repository and how to pass args. How do you see it?

Reply:
Hi @adriangonz ,

My apologies for my delay too, day-to-day!

I see both sides of the solution, if you prefer to keep the parameter, no big deal, I can keep it. IMO it can be confusing to have a parameter that the implementation doesn't always use. I will update code with your choice.

Reply:
Done